snippets/terraform/deployments/create-or-update/main.tf

```diff
@@ -0,0 +1,96 @@
+terraform {
```

Comment:
Could we move some of this prerequisite code for the resource group/networking to a prerequisites folder like we have for the ARM/Bicep templates and then import it into this example? Just to keep the code snippets modular and limited to just what we are trying to show the user how to do?

Reply:
yep doable. I will have a look at the ARM code to see what we can do here.

Reply:
@agazeley I am going to push more code in a follow up where I will DRY out the common code.

P.S I am going to leverage the common code/prerequisite code within the various resources so the pattern will be a little different than arm-templates.